Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp

```diff
@@ -164,4 +167,90 @@ Expected<std::pair<String, Vector<IntSize>>, ImageDecodingError> utiAndAvailable
     return std::make_pair(WTFMove(uti), WTFMove(sizes));
 }
 
+static RefPtr<NativeImage> createNativeImageFromData(std::span<const uint8_t> data, std::optional<FloatSize> preferredSize)
+{
+    Ref buffer = FragmentedSharedBuffer::create(data);
+    Ref imageDecoder = ImageDecoderCG::create(buffer.get(), AlphaOption::Premultiplied, GammaAndColorProfileOption::Applied);
+    imageDecoder->setData(buffer.get(), true);
+    if (imageDecoder->encodedDataStatus() == EncodedDataStatus::Error)
+        return nullptr;
+
+    auto sourceUTI = imageDecoder->uti();
+    if (!isSupportedImageType(sourceUTI))
+        return nullptr;
+
+    size_t frameCount = imageDecoder->frameCount();
+    auto preferredIndex = [&]() -> size_t {
+        if (!preferredSize)
+            return imageDecoder->primaryFrameIndex();
+        for (size_t index = 0; index < frameCount; ++index) {
+            if (imageDecoder->frameSizeAtIndex(index) == *preferredSize)
+                return index;
+        }
+        return imageDecoder->primaryFrameIndex();
+    };
+    RetainPtr image = imageDecoder->createFrameImageAtIndex(preferredIndex());
+    if (!image)
+        return nullptr;
+
+    return NativeImage::create(WTFMove(image));
+}
+
+static RefPtr<SharedBuffer> expandNativeImageToData(NativeImage& image, ASCIILiteral uti, std::span<const unsigned> lengths)
+{
+    RetainPtr colorSpace = adoptCF(CGColorSpaceCreateWithName(kCGColorSpaceGenericRGB));
```

Comment:
Why is this the correct color space? Why not sRGB?

Reply:
sRGB should work too; I will change.